rllib/execution/parallel_requests.py

```diff
@@ -182,3 +184,85 @@ def wait_asynchronous_requests(
         remote_fn=None,
     )
     return ready_requests
+
+
+class AsyncRequestsManager:
```

Comment:
I like that we are packaging this rather complex behavior into a nice utility class! Especially given that with the current solution, we require the user to carry around the state of this (currently e.g. `Trainer._remote_requests_in_flight`).

Reply:
Could you write a few more lines on how this thing works under the hood? Like what state is kept and what's a typical synopsis of using this class (maybe 1 Example snippet)?